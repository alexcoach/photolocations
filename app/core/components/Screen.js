// @flow

import React from 'react';
import { connect } from 'react-redux';
import { StyleSheet } from 'react-native';

import {
    Container, Header, Body,
    Content, Thumbnail, Left, Right,
    Title, Tab, Tabs, TabHeading, Icon, Text, Button,
} from 'native-base';

import { closeModal } from 'core/actions/Modals';
import { Navigation } from 'core/services';

import theme from 'app/theme';

export type TabConfig = {
    icon?: string | Object,
    title?: any | string,
    content: any,
};

type Props = {
    title?: string,
    left?: {
        type?: string,
        icon?: Object,
    },
    right?: any,
    tabs?: Array<TabConfig>,
    dispatch: Function,
    children?: any,
};

class Screen extends React.PureComponent<Props>{
    renderHeaderLeft = () => {
        const { left } = this.props;
        if (!left) {
            return null;
        }
        const { type } = left;
        if (type === 'logo') {
            return (
                <Thumbnail square small source={require('assets/app-icon.png')} />
            );
        }
        if (type === 'back') {
            return (
                <Button transparent onPress={() => Navigation.goBack()}>
                    <Icon type="Feather" name="chevron-left" />
                </Button>
            );
        }
        if (type === 'closeModal') {
            return (
                <Button
                    transparent
                    onPress={() => {
                        this.props.dispatch(closeModal());
                    }}
                >
                    <Icon type="Feather" name="chevron-left" {...left.icon} />
                </Button>
            );
        }
        return null;
    };
    renderHeader = () => {
        const { tabs, title, right } = this.props;
        return (
            <Header hasTabs={!!(tabs && tabs.length)} >
                <Left>
                    {this.renderHeaderLeft()}
                </Left>
                <Body>
                    <Title>{title}</Title>
                </Body>
                <Right>{right}</Right>
            </Header>
        );
    };
    renderTabHeadingIcon = (tabConfig: TabConfig) => {
        if (!tabConfig.icon) {
            return null;
        }
        const iconProps = (typeof tabConfig.icon === 'string') ? { name: tabConfig.icon } : tabConfig.icon;
        return <Icon {...iconProps} />;

    };
    renderTabHeading = (tabConfig: TabConfig) => (
        <TabHeading>
            {this.renderTabHeadingIcon(tabConfig)}
            <Text>{tabConfig.title}</Text>
        </TabHeading>
    );
    renderTab = (tabConfig: TabConfig, index: number) => (
        <Tab key={index} heading={this.renderTabHeading(tabConfig)}>
            {tabConfig.content}
        </Tab>
    );
    renderTabs = () => {
        const { tabs } = this.props;
        if (!tabs || !tabs.length) {
            return null;
        }
        return (
            <Tabs>
                {tabs.map(this.renderTab)}
            </Tabs>
        );
    };
    renderContent = () => {
        const { children } = this.props;
        if (!children) {
            return null;
        }
        return <Content>{children}</Content>;
    };
    render() {
        return (
            <Container style={styles.container}>
                {this.renderHeader()}
                {this.renderTabs()}
                {this.renderContent()}
            </Container>
        );
    }
}

export default connect()(Screen);

const styles = StyleSheet.create({
    container: {
        backgroundColor: theme.brandLight,
    },
});
