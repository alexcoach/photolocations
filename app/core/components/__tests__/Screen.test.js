import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import { Text } from 'native-base';
import Screen from '../Screen';

const mockStore = configureStore([]);

const state = {};

it('<Screen> with children renders correctly', () => {
    const wrapper = shallow(
        <Screen
            title="Test Screen"
            left={{ type: 'back', icon: { name: 'home' }}}
        >
            <Text>Children</Text>
        </Screen>,
        { context: { store: mockStore(state) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
});

it('<Screen> with tabs renders correctly', () => {
    const wrapper = shallow(
        <Screen
            title="Test Screen"
            left={{ type: 'back', icon: { name: 'home' }}}
            tabs={[
                { icon: 'home', title: 'Tab1', content: <Text>Tab1 Content</Text> },
                { icon: 'home', title: 'Tab2', content: <Text>Tab2 Content</Text> },
            ]}
        />,
        { context: { store: mockStore(state) } },
    );
    expect(wrapper.dive()).toMatchSnapshot();
});
