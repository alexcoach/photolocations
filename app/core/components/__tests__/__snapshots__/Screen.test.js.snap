// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Screen> with children renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Screen
    dispatch={[Function]}
    left={
      Object {
        "icon": Object {
          "name": "home",
        },
        "type": "back",
      }
    }
    title="Test Screen"
  >
    <Styled(Text)>
      Children
    </Styled(Text)>
  </Screen>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Header)
          hasTabs={false}
        >
          <Styled(Left)>
            <Styled(Button)
              onPress={[Function]}
              transparent={true}
            >
              <Styled(Icon)
                name="chevron-left"
                type="Feather"
              />
            </Styled(Button)>
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Title)>
              Test Screen
            </Styled(Title)>
          </Styled(Body)>
          <Styled(Right) />
        </Styled(Header)>,
        null,
        <Styled(Content)>
          <Styled(Text)>
            Children
          </Styled(Text)>
        </Styled(Content)>,
      ],
      "style": Object {
        "backgroundColor": "#f4f4f4",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Left)>
              <Styled(Button)
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />
              </Styled(Button)>
            </Styled(Left)>,
            <Styled(Body)>
              <Styled(Title)>
                Test Screen
              </Styled(Title)>
            </Styled(Body)>,
            <Styled(Right) />,
          ],
          "hasTabs": false,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Button)
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />
              </Styled(Button)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />,
                "onPress": [Function],
                "transparent": true,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "chevron-left",
                  "type": "Feather",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Title)>
                Test Screen
              </Styled(Title)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test Screen",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Test Screen",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Styled(Text)>
            Children
          </Styled(Text)>,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Children",
            "virtual": undefined,
          },
          "ref": null,
          "rendered": "Children",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Header)
            hasTabs={false}
          >
            <Styled(Left)>
              <Styled(Button)
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />
              </Styled(Button)>
            </Styled(Left)>
            <Styled(Body)>
              <Styled(Title)>
                Test Screen
              </Styled(Title)>
            </Styled(Body)>
            <Styled(Right) />
          </Styled(Header)>,
          null,
          <Styled(Content)>
            <Styled(Text)>
              Children
            </Styled(Text)>
          </Styled(Content)>,
        ],
        "style": Object {
          "backgroundColor": "#f4f4f4",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Left)>
                <Styled(Button)
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Icon)
                    name="chevron-left"
                    type="Feather"
                  />
                </Styled(Button)>
              </Styled(Left)>,
              <Styled(Body)>
                <Styled(Title)>
                  Test Screen
                </Styled(Title)>
              </Styled(Body)>,
              <Styled(Right) />,
            ],
            "hasTabs": false,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Button)
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Icon)
                    name="chevron-left"
                    type="Feather"
                  />
                </Styled(Button)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Icon)
                    name="chevron-left"
                    type="Feather"
                  />,
                  "onPress": [Function],
                  "transparent": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "chevron-left",
                    "type": "Feather",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Title)>
                  Test Screen
                </Styled(Title)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Test Screen",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Test Screen",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Text)>
              Children
            </Styled(Text)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Children",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Children",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;

exports[`<Screen> with tabs renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Screen
    dispatch={[Function]}
    left={
      Object {
        "icon": Object {
          "name": "home",
        },
        "type": "back",
      }
    }
    tabs={
      Array [
        Object {
          "content": <Styled(Text)>
            Tab1 Content
          </Styled(Text)>,
          "icon": "home",
          "title": "Tab1",
        },
        Object {
          "content": <Styled(Text)>
            Tab2 Content
          </Styled(Text)>,
          "icon": "home",
          "title": "Tab2",
        },
      ]
    }
    title="Test Screen"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(Header)
          hasTabs={true}
        >
          <Styled(Left)>
            <Styled(Button)
              onPress={[Function]}
              transparent={true}
            >
              <Styled(Icon)
                name="chevron-left"
                type="Feather"
              />
            </Styled(Button)>
          </Styled(Left)>
          <Styled(Body)>
            <Styled(Title)>
              Test Screen
            </Styled(Title)>
          </Styled(Body)>
          <Styled(Right) />
        </Styled(Header)>,
        <ScrollableTabView
          contentProps={Object {}}
          initialPage={0}
          locked={false}
          onChangeTab={[Function]}
          onScroll={[Function]}
          page={-1}
          prerenderingSiblingsNumber={0}
          scrollWithoutAnimation={false}
          tabBarPosition="top"
        >
          <Styled(Tab)
            heading={
              <Styled(TabHeadingNB)>
                <Styled(Icon)
                  name="home"
                />
                <Styled(Text)>
                  Tab1
                </Styled(Text)>
              </Styled(TabHeadingNB)>
            }
          >
            <Styled(Text)>
              Tab1 Content
            </Styled(Text)>
          </Styled(Tab)>
          <Styled(Tab)
            heading={
              <Styled(TabHeadingNB)>
                <Styled(Icon)
                  name="home"
                />
                <Styled(Text)>
                  Tab2
                </Styled(Text)>
              </Styled(TabHeadingNB)>
            }
          >
            <Styled(Text)>
              Tab2 Content
            </Styled(Text)>
          </Styled(Tab)>
        </ScrollableTabView>,
        null,
      ],
      "style": Object {
        "backgroundColor": "#f4f4f4",
      },
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Left)>
              <Styled(Button)
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />
              </Styled(Button)>
            </Styled(Left)>,
            <Styled(Body)>
              <Styled(Title)>
                Test Screen
              </Styled(Title)>
            </Styled(Body)>,
            <Styled(Right) />,
          ],
          "hasTabs": true,
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Button)
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />
              </Styled(Button)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />,
                "onPress": [Function],
                "transparent": true,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "name": "chevron-left",
                  "type": "Feather",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Title)>
                Test Screen
              </Styled(Title)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test Screen",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Test Screen",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Tab)
              heading={
                <Styled(TabHeadingNB)>
                  <Styled(Icon)
                    name="home"
                  />
                  <Styled(Text)>
                    Tab1
                  </Styled(Text)>
                </Styled(TabHeadingNB)>
              }
            >
              <Styled(Text)>
                Tab1 Content
              </Styled(Text)>
            </Styled(Tab)>,
            <Styled(Tab)
              heading={
                <Styled(TabHeadingNB)>
                  <Styled(Icon)
                    name="home"
                  />
                  <Styled(Text)>
                    Tab2
                  </Styled(Text)>
                </Styled(TabHeadingNB)>
              }
            >
              <Styled(Text)>
                Tab2 Content
              </Styled(Text)>
            </Styled(Tab)>,
          ],
          "contentProps": Object {},
          "initialPage": 0,
          "locked": false,
          "onChangeTab": [Function],
          "onScroll": [Function],
          "page": -1,
          "prerenderingSiblingsNumber": 0,
          "scrollWithoutAnimation": false,
          "tabBarPosition": "top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Text)>
                Tab1 Content
              </Styled(Text)>,
              "heading": <Styled(TabHeadingNB)>
                <Styled(Icon)
                  name="home"
                />
                <Styled(Text)>
                  Tab1
                </Styled(Text)>
              </Styled(TabHeadingNB)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Tab1 Content",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Tab1 Content",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Text)>
                Tab2 Content
              </Styled(Text)>,
              "heading": <Styled(TabHeadingNB)>
                <Styled(Icon)
                  name="home"
                />
                <Styled(Text)>
                  Tab2
                </Styled(Text)>
              </Styled(TabHeadingNB)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Tab2 Content",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Tab2 Content",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Header)
            hasTabs={true}
          >
            <Styled(Left)>
              <Styled(Button)
                onPress={[Function]}
                transparent={true}
              >
                <Styled(Icon)
                  name="chevron-left"
                  type="Feather"
                />
              </Styled(Button)>
            </Styled(Left)>
            <Styled(Body)>
              <Styled(Title)>
                Test Screen
              </Styled(Title)>
            </Styled(Body)>
            <Styled(Right) />
          </Styled(Header)>,
          <ScrollableTabView
            contentProps={Object {}}
            initialPage={0}
            locked={false}
            onChangeTab={[Function]}
            onScroll={[Function]}
            page={-1}
            prerenderingSiblingsNumber={0}
            scrollWithoutAnimation={false}
            tabBarPosition="top"
          >
            <Styled(Tab)
              heading={
                <Styled(TabHeadingNB)>
                  <Styled(Icon)
                    name="home"
                  />
                  <Styled(Text)>
                    Tab1
                  </Styled(Text)>
                </Styled(TabHeadingNB)>
              }
            >
              <Styled(Text)>
                Tab1 Content
              </Styled(Text)>
            </Styled(Tab)>
            <Styled(Tab)
              heading={
                <Styled(TabHeadingNB)>
                  <Styled(Icon)
                    name="home"
                  />
                  <Styled(Text)>
                    Tab2
                  </Styled(Text)>
                </Styled(TabHeadingNB)>
              }
            >
              <Styled(Text)>
                Tab2 Content
              </Styled(Text)>
            </Styled(Tab)>
          </ScrollableTabView>,
          null,
        ],
        "style": Object {
          "backgroundColor": "#f4f4f4",
        },
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Left)>
                <Styled(Button)
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Icon)
                    name="chevron-left"
                    type="Feather"
                  />
                </Styled(Button)>
              </Styled(Left)>,
              <Styled(Body)>
                <Styled(Title)>
                  Test Screen
                </Styled(Title)>
              </Styled(Body)>,
              <Styled(Right) />,
            ],
            "hasTabs": true,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Button)
                  onPress={[Function]}
                  transparent={true}
                >
                  <Styled(Icon)
                    name="chevron-left"
                    type="Feather"
                  />
                </Styled(Button)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Styled(Icon)
                    name="chevron-left"
                    type="Feather"
                  />,
                  "onPress": [Function],
                  "transparent": true,
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "name": "chevron-left",
                    "type": "Feather",
                    "virtual": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Title)>
                  Test Screen
                </Styled(Title)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Test Screen",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Test Screen",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": undefined,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Styled(Tab)
                heading={
                  <Styled(TabHeadingNB)>
                    <Styled(Icon)
                      name="home"
                    />
                    <Styled(Text)>
                      Tab1
                    </Styled(Text)>
                  </Styled(TabHeadingNB)>
                }
              >
                <Styled(Text)>
                  Tab1 Content
                </Styled(Text)>
              </Styled(Tab)>,
              <Styled(Tab)
                heading={
                  <Styled(TabHeadingNB)>
                    <Styled(Icon)
                      name="home"
                    />
                    <Styled(Text)>
                      Tab2
                    </Styled(Text)>
                  </Styled(TabHeadingNB)>
                }
              >
                <Styled(Text)>
                  Tab2 Content
                </Styled(Text)>
              </Styled(Tab)>,
            ],
            "contentProps": Object {},
            "initialPage": 0,
            "locked": false,
            "onChangeTab": [Function],
            "onScroll": [Function],
            "page": -1,
            "prerenderingSiblingsNumber": 0,
            "scrollWithoutAnimation": false,
            "tabBarPosition": "top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Text)>
                  Tab1 Content
                </Styled(Text)>,
                "heading": <Styled(TabHeadingNB)>
                  <Styled(Icon)
                    name="home"
                  />
                  <Styled(Text)>
                    Tab1
                  </Styled(Text)>
                </Styled(TabHeadingNB)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Tab1 Content",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Tab1 Content",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <Styled(Text)>
                  Tab2 Content
                </Styled(Text)>,
                "heading": <Styled(TabHeadingNB)>
                  <Styled(Icon)
                    name="home"
                  />
                  <Styled(Text)>
                    Tab2
                  </Styled(Text)>
                </Styled(TabHeadingNB)>,
                "virtual": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Tab2 Content",
                  "virtual": undefined,
                },
                "ref": null,
                "rendered": "Tab2 Content",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
